cmake/app/boilerplate.cmake

```diff
@@ -532,6 +534,84 @@ set_property(TARGET ${IMAGE}app PROPERTY ARCHIVE_OUTPUT_DIRECTORY ${IMAGE}app)
 
 add_subdirectory(${ZEPHYR_BASE} ${__build_dir})
 
+if(EXISTS ${APPLICATION_SOURCE_DIR}/pm.yml)
+  zephyr_get_include_directories_for_lang(C current_includes)
+  get_property(current_defines GLOBAL PROPERTY PROPERTY_LINKER_SCRIPT_DEFINES)
+  set(pre_partition_mananger_conf
+    ${PROJECT_BINARY_DIR}/include/generated/pm.yml.pre)
+  set(partition_manager_conf_target ${IMAGE}pm_conf_target)
+  add_custom_command(
+    OUTPUT ${pre_partition_mananger_conf}
+    COMMAND ${CMAKE_C_COMPILER}
+    -x assembler-with-cpp
+    ${NOSTDINC_F}
+    ${NOSYSDEF_CFLAG}
+    ${current_includes}
+    ${current_defines}
+    -E ${APPLICATION_SOURCE_DIR}/pm.yml
+    -P # Prevent generation of debug `#line' directives.
+    ${partition_manager_override_include}
+    -o ${pre_partition_mananger_conf}
+    VERBATIM
+    WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+    )
+  add_custom_target(
+    ${partition_manager_conf_target}
+    DEPENDS
+    ${pre_partition_mananger_conf}
+    )
+  set_property(
+    GLOBAL APPEND PROPERTY
+    PARTITION_MANAGER_CONFIG_FILES
+    ${pre_partition_mananger_conf}
+    )
+  set_property(
+    GLOBAL APPEND PROPERTY
+    PARTITION_MANAGER_CONFIG_TARGETS
+    ${partition_manager_conf_target}
+    )
+endif()
+
+if(FIRST_BOILERPLATE_EXECUTION)
+  get_property(
+      partition_manager_config_files
+    GLOBAL PROPERTY
+    PARTITION_MANAGER_CONFIG_FILES
+    )
+  print(partition_manager_config_files)
+  if(partition_manager_config_files)
+    get_property(
+      partition_manager_config_targets
+      GLOBAL PROPERTY
+      PARTITION_MANAGER_CONFIG_TARGETS
+      )
+    file(GLOB_RECURSE autoconf_files "${PROJECT_BINARY_DIR}/**/autoconf.h")
```

Comment:
globbing like this should be avoided.

Better to iterate over image build dirs.

Reply:
why should it be avoided? How should I populate the list of all autoconf.h files if not by globbing?

Reply:
Refactored the script to not need these files, so this is now removed as requested.